source/ptreferential/ptreferential_ng.cpp

```diff
@@ -291,25 +292,31 @@ struct Eval : boost::static_visitor<Indexes> {
                         indexes.insert(l->idx);
                 }
             }
-        } else if (f.type == "disruption"
+        } else if (type == Type_e::Impact
                    && ((f.method == "tag" && f.args.size() == 1) || (f.method == "tags" && f.args.size() >= 1))) {
             indexes = get_impacts_by_tags(f.args, data);
-        } else if ((f.method == "since" && f.args.size() == 2) || (f.method == "until" && f.args.size() == 2)
-                   || (f.method == "between" && f.args.size() == 3)) {
+        }
+        // since/until/between are available for disruption and vehicle_journey,
+        // but VJ requires the additional param `data_freshness`
+        // so below grammar provides the following methods:
+        // * disruption.between(since, until)
+        // * vehicle_journey.between(since, until, data_freshness)
+        else if ((f.method == "since" && f.args.size() == 1u + nb_extra_args_between(type))
+                 || (f.method == "until" && f.args.size() == 1u + nb_extra_args_between(type))
+                 || (f.method == "between" && f.args.size() == 2u + nb_extra_args_between(type))) {
             boost::optional<boost::posix_time::ptime> since, until;
-            auto rt_level = type::RTLevel::Base;
             if (f.method == "since") {
                 since = from_datetime(f.args.at(0));
-                rt_level = rt_level_from_string(f.args.at(1));
             } else if (f.method == "until") {
                 until = from_datetime(f.args.at(0));
-                rt_level = rt_level_from_string(f.args.at(1));
             } else if (f.method == "between") {
                 since = from_datetime(f.args.at(0));
                 until = from_datetime(f.args.at(1));
-                rt_level = rt_level_from_string(f.args.at(2));
             }
-            const auto type = type_by_caption(f.type);
+            auto rt_level = type::RTLevel::Base;  // useful only for VJ
```

Comment:
`auto const rt_level = type == Type_e::VehicleJourney ? type::RTLevel::Base : rt_level_from_string(f.args.back())`

`rt_level` can be const this way.